src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/LockedDownGerritEventTest.java

```diff
@@ -0,0 +1,147 @@
+
+import static com.sonymobile.tools.gerrit.gerritevents.mock.SshdServerMock.GERRIT_STREAM_EVENTS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import jenkins.model.Jenkins;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.model.User;
+import hudson.security.GlobalMatrixAuthorizationStrategy;
+import hudson.security.HudsonPrivateSecurityRealm;
+
+import org.apache.sshd.SshServer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.GerritCause;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.mock.DuplicatesUtil;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.mock.Setup;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.mock.TestUtils;
+import com.sonymobile.tools.gerrit.gerritevents.mock.SshdServerMock;
+
+/**
+ * Unit test to ensure build can be triggered even if
+ * security is enabled and only authenticated users can login.
+ *
+ * @author Scott Hebert &lt;scott.hebert@ericsson.com&gt;
+ */
+
+public class LockedDownGerritEventTest {
+
+    /**
+     * An instance of Jenkins Rule.
+     */
+    // CS IGNORE VisibilityModifier FOR NEXT 2 LINES. REASON: JenkinsRule.
+    @Rule
+    public final JenkinsRule j = new JenkinsRule();
+
+    private final String gerritServerName = "testServer";
+    private final String projectName = "testProject";
+    private final int port = 29418;
+
+    private SshdServerMock server;
+    private SshServer sshd;
+
+    /**
+     * Runs before test method.
+     *
+     * @throws Exception throw if so.
+     */
+    @Before
+    public void setUp() throws Exception {
+        SshdServerMock.generateKeyPair();
+        server = new SshdServerMock();
+        sshd = SshdServerMock.startServer(port, server);
+        server.returnCommandFor("gerrit ls-projects", SshdServerMock.EofCommandMock.class);
+        server.returnCommandFor(GERRIT_STREAM_EVENTS, SshdServerMock.CommandMock.class);
+        server.returnCommandFor("gerrit review.*", SshdServerMock.EofCommandMock.class);
+        server.returnCommandFor("gerrit version", SshdServerMock.EofCommandMock.class);
+    }
+
+    /**
+     * Runs after test method.
+     *
+     * @throws Exception throw if so.
+     */
+    @After
+    public void tearDown() throws Exception {
+        sshd.stop(true);
+        sshd = null;
+    }
+
+    /**
+     * Lock down the instance.
+     * @throws Exception throw if so.
+     */
+    private void lockDown() throws Exception {
+        GlobalMatrixAuthorizationStrategy auth = new GlobalMatrixAuthorizationStrategy();
+        j.jenkins.setAuthorizationStrategy(auth);
+        j.jenkins.setCrumbIssuer(null);
+        HudsonPrivateSecurityRealm realm = new HudsonPrivateSecurityRealm(false);
```

Comment:
Example of usage:
https://github.com/jenkinsci/build-failure-analyzer-plugin/blob/master/src/test/java/com/sonyericsson/jenkins/plugins/bfa/CauseManagementPermissionTest.java#L40